api/envoy/config/listener/v3/listener.proto

```diff
@@ -90,6 +90,14 @@ message Listener {
   // Configuration for envoy internal listener. All the future internal listener features should be added here.
   // [#not-implemented-hide:]
   message InternalListenerConfig {
+    // The max contiguous chained internal listener to internal cluster is allowed. 0 means upstream
```

Comment:
Apart from the comment threads below:
1) I would rather we just add this with the feature so we can see how it works at that time.
2) This text needs significant wordsmithing but we can do that once we agree on the final implementation and design.

Reply:
Got it. Thanks!